README.md

```diff
@@ -20,9 +20,390 @@ Or install it yourself as:
+| `secret`       | Your Cassia API Developer Secret                                                                                                                                                           |
+ `7d8b99e989344b6f` |
+
+### Setting your configuration
+
+To set configuration options use the `Cassia.configure` method:
+
+```ruby
+Cassia.configure do |config|
+  config.ac_url = ENV['CASSIA_AC_URL']
+  config.client_id = ENV['CASSIA_CLIENT_ID']
+  config.secret = ENV['CASSIA_SECRET']
+end
+```
+
+## Resources
+
+### AccessControllers
+
+#### Retrieve An Access Token From the Cassia API
+```ruby
+token = access_controller.get_token
+```
+`access_controller` is a `Cassia::AccessController` object which contains attributes and structures of a Cassia access controller. 
+
+`get_token` will return the `access_token` attribute of `access_controller` if it is not `nil`. Otherwise, it will generate an access token encoded from `config.client_id` and `config.secret` using base64. 
+
+#### Obtain Cassia router's status
+```ruby
+routers = access_controller.get_all_routers_status
+```
+`access_controller` is a `Cassia::AccessController` object which contains attributes and structures of a Cassia access controller. 
+
+`routers` is an Array of `Router` objects connected to `access_controller`.
+
+#### Switch Router Auto-Selection
+```ruby
+access_controller.switch_autoselect(flag: 1)
+```
+`access_controller` is a `Cassia::AccessController` object which contains attributes and structures of a Cassia access controller. 
+
+If `flag` is 1, the router auto-selection function will be enabled. If `flag` is 0, the router auto-selection function will be disabled.
+
+#### Open Scanning for all the routers in the router list
+```ruby
+access_controller.open_scan(aps: ["CC:1B:E0:E7:FE:F8",  "CC:1B:E0:E7:FE:F8", "CC:1B:E0:E7:FE:F8"], chip: 0, active: 0, filter_name: "cassia", filter_mac: "CC:1B:E0:E7:FE:F7", filter_uuid: "00001800-0000-1000-800000805f9b34fb")
+```
+`access_controller` is a `Cassia::AccessController` object which contains attributes and structures of a Cassia access controller. 
+
+`aps` is an array of one or multiple router's MAC address.
+`chip` (optional) means which chip to scan.
+`active` (optional) is either 0 or 1. 0 means enable passive scanning, 1 means enable active scanning.
+`filter_name` (optional) is a filter for device name.
+`filter_mac` (optional) is a filter for device MAC.
+`filter_uuid` (optional) is a filter for device UUID.
+
+#### Close Scanning for all the routers in the router list
+```ruby
+access_controller.close_scan(aps: ["CC:1B:E0:E7:FE:F8",  "CC:1B:E0:E7:FE:F8", "CC:1B:E0:E7:FE:F8"])
+```
+`access_controller` is a `Cassia::AccessController` object which contains attributes and structures of a Cassia access controller.
+ 
+`aps` is an array of one or multiple router's MAC address.
+
+#### Connect to One Automatically Selected Router from the router list
```

Comment:
this is actually connecting a device to one router automatically from you list of routers